File: app/src/main/java/com/progix/fridgex/light/adapter/recipe/InfoAdapter.java
package com.progix.fridgex.light.adapter.recipe;

import static android.provider.AlarmClock.ACTION_SET_TIMER;
import static android.provider.AlarmClock.EXTRA_LENGTH;
import static android.provider.AlarmClock.EXTRA_MESSAGE;
import static android.provider.AlarmClock.EXTRA_SKIP_UI;
import static android.view.View.GONE;
import static android.view.View.VISIBLE;
import static com.progix.fridgex.light.activity.MainActivity.mDb;
import static com.progix.fridgex.light.activity.SecondActivity.name;

import android.content.Context;
import android.content.Intent;
import android.database.Cursor;
import android.net.Uri;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.google.android.material.dialog.MaterialAlertDialogBuilder;
import com.progix.fridgex.light.R;
import com.progix.fridgex.light.activity.SecondActivity;
import com.progix.fridgex.light.model.InfoItem;

import java.util.ArrayList;
import java.util.Objects;


public class InfoAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {

    Context context;
    ArrayList<InfoItem> infoList;

    public InfoAdapter(Context context, ArrayList<InfoItem> infoList) {
        this.context = context;
        this.infoList = infoList;
    }

    @Override
    public int getItemViewType(int position) {
        if (position == 7) return 2;
        else return 1;
    }

    @NonNull
    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        if (viewType == 1) {
            return new ViewHolder(
                    LayoutInflater.from(parent.getContext()).inflate(R.layout.item_info, parent, false)
            );
        } else {
            return new ViewHolder2(
                    LayoutInflater.from(parent.getContext()).inflate(R.layout.item_info_7, parent, false)
            );
        }
    }

    @Override
    public void onBindViewHolder(@NonNull RecyclerView.ViewHolder holderMain, int position) {
        if (holderMain.getItemViewType() == 1) {
            ViewHolder holder = (ViewHolder) holderMain;
            holder.itemView.setClickable(false);
            if (position == 0) {
                holder.itemView.setOnClickListener(v -> {
                    new MaterialAlertDialogBuilder(context, R.style.modeAlert)
                            .setTitle(context.getString(R.string.timer))
                            .setMessage(context.getString(R.string.setTimer) + SecondActivity.name)
                            .setPositiveButton(context.getString(R.string.cont), (b, z) -> {
                                Intent intent = new Intent(ACTION_SET_TIMER);
                                intent.putExtra(
                                        EXTRA_LENGTH,
                                        Integer.parseInt(infoList.get(position).getValue().split(" ")[0]) * 60
                                );
                                intent.putExtra(EXTRA_SKIP_UI, false);
                                intent.putExtra(EXTRA_MESSAGE, SecondActivity.name);
                                context.startActivity(intent);
                            })
                            .setNegativeButton(context.getString(R.string.cancel), null)
                            .show();
                });
            }

            if (Objects.equals(infoList.get(position).getValue(), "0"))
                holder.itemView.setVisibility(GONE);
            holder.name.setText(infoList.get(position).getName());
            holder.value.setText(infoList.get(position).getValue());
            holder.image.setImageResource(infoList.get(position).getImage());
            if (Objects.equals(infoList.get(position).getName(), context.getString(R.string.source)) && !Objects.equals(infoList.get(position).getValue(), "Авторский")) {
                holder.itemView.setOnClickListener(v -> new MaterialAlertDialogBuilder(context, R.style.modeAlert)
                        .setTitle(context.getString(R.string.redirect))
                        .setMessage(context.getString(R.string.redirectMessage))
                        .setPositiveButton(context.getString(R.string.cont), (x, y) -> {
                            String url = "http://" + infoList.get(position).getValue();
                            Intent browserIntent = new Intent(Intent.ACTION_VIEW, Uri.parse(url));
                            context.startActivity(browserIntent);
                        })
                        .setNegativeButton(context.getString(R.string.cancel), null)
                        .show());
            } else if (Objects.equals(infoList.get(position).getValue(), "Авторский")) {
                holder.image.setImageResource(R.drawable.ic_round_edit_24);
            }
        } else {
            ViewHolder2 holder = (ViewHolder2) holderMain;
            holder.itemView.setClickable(false);

            if (position == 0) {
                holder.itemView.setOnClickListener(v -> new MaterialAlertDialogBuilder(context, R.style.modeAlert)
                        .setTitle(context.getString(R.string.timer))
                        .setMessage(context.getString(R.string.setTimer) + SecondActivity.name)
                        .setPositiveButton(context.getString(R.string.cont), (x, y) -> {
                            Intent intent = new Intent(ACTION_SET_TIMER);
                            intent.putExtra(
                                    EXTRA_LENGTH,
                                    Integer.parseInt(infoList.get(position).getValue().split(" ")[0]) * 60
                            );
                            intent.putExtra(EXTRA_SKIP_UI, false);
                            intent.putExtra(EXTRA_MESSAGE, SecondActivity.name);
                            context.startActivity(intent);
                        })
                        .setNegativeButton(context.getString(R.string.cancel), null)
                        .show());
            }

            holder.name.setText(infoList.get(position).getName());
            holder.value.setText(infoList.get(position).getValue());
            holder.image.setImageResource(infoList.get(position).getImage());
            if (Objects.equals(infoList.get(position).getName(), context.getString(R.string.source)) && !Objects.equals(infoList.get(position).getValue(), "Авторский")) {
                holder.itemView.setOnClickListener(v -> {
                    new MaterialAlertDialogBuilder(context, R.style.modeAlert)
                            .setTitle(context.getString(R.string.redirect))
                            .setMessage(context.getString(R.string.redirectMessage))
                            .setPositiveButton(
                                    context.getString(R.string.cont), (x, y) -> {
                                        String url = "http://" + infoList.get(position).getValue();
                                        Intent browserIntent = new Intent(Intent.ACTION_VIEW, Uri.parse(url));
                                        context.startActivity(browserIntent);
                                    })
                            .setNegativeButton(context.getString(R.string.cancel), null)
                            .show();
                });
            } else if (Objects.equals(infoList.get(position).getValue(), "Авторский")) {
                holder.image.setImageResource(R.drawable.ic_round_edit_24);
            }
            if (position == 7) {
                Cursor cursor = mDb.rawQuery("SELECT * FROM recipes WHERE id = ?", new String[]{infoList.get(position).getValue()});
                cursor.moveToFirst();
                boolean starred = cursor.getInt(7) == 1;
                boolean banned = cursor.getInt(14) == 1;
                if (starred && banned) {
                    holder.banLayout.setVisibility(VISIBLE);
                    holder.starLayout.setVisibility(VISIBLE);
                    holder.itemView.setVisibility(VISIBLE);
                    holder.value.setText(context.getString(R.string.inStarred));
                    holder.value2.setText(context.getString(R.string.inBanned));
                } else if (starred && !banned) {
                    holder.value.setText(context.getString(R.string.inStarred));
                    holder.banLayout.setVisibility(GONE);
                    holder.starLayout.setVisibility(VISIBLE);
                    holder.itemView.setVisibility(VISIBLE);
                } else if (!starred && banned) {
                    holder.starLayout.setVisibility(GONE);
                    holder.banLayout.setVisibility(VISIBLE);
                    holder.itemView.setVisibility(VISIBLE);
                    holder.value2.setText(context.getString(R.string.inBanned));
                } else if (!starred && !banned) holder.itemView.setVisibility(GONE);
                cursor.close();
            }
        }
    }

    @Override
    public int getItemCount() {
        return infoList.size();
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {
        TextView name;
        TextView value;
        ImageView image;

        public ViewHolder(@NonNull View view) {
            super(view);
            name = view.findViewById(R.id.name);
            value = view.findViewById(R.id.value);
            image = view.findViewById(R.id.image);
        }

    }

    public static class ViewHolder2 extends RecyclerView.ViewHolder {
        TextView name;
        TextView value;
        ImageView image;
        TextView value2;
        LinearLayout starLayout;
        LinearLayout banLayout;

        public ViewHolder2(@NonNull View view) {
            super(view);
            name = view.findViewById(R.id.name);
            value = view.findViewById(R.id.value);
            image = view.findViewById(R.id.star);
            value2 = view.findViewById(R.id.value2);
            starLayout = view.findViewById(R.id.star_layout);
            banLayout = view.findViewById(R.id.ban_layout);
        }
    }

    @Override
    public void onViewRecycled(RecyclerView.ViewHolder holder) {
        super.onViewRecycled(holder);
        holder.itemView.setOnClickListener(null);
    }

}

File: app/src/main/java/com/progix/fridgex/light/adapter/recipe/InfoAdapter.kt
package com.progix.fridgex.light.adapter.recipe

import android.content.Context
import android.content.Intent
import android.database.Cursor
import android.net.Uri
import android.provider.AlarmClock.*
import android.view.LayoutInflater
import android.view.View
import android.view.View.GONE
import android.view.View.VISIBLE
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.LinearLayout
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.google.android.material.dialog.MaterialAlertDialogBuilder
import com.progix.fridgex.light.R
import com.progix.fridgex.light.activity.MainActivity.Companion.mDb
import com.progix.fridgex.light.activity.SecondActivity
import com.progix.fridgex.light.model.InfoItem


class InfoAdapter(var context: Context, private var infoList: ArrayList<InfoItem>) :
    RecyclerView.Adapter<RecyclerView.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
        return when (viewType) {
            1 -> {
                ViewHolder(
                    LayoutInflater.from(parent.context).inflate(R.layout.item_info, parent, false)
                )
            }
            else -> {
                ViewHolder2(
                    LayoutInflater.from(parent.context).inflate(R.layout.item_info_7, parent, false)
                )
            }
        }
    }


    override fun getItemViewType(position: Int): Int {
        return when (position) {
            7 -> 2
            else -> 1
        }
    }

    override fun onBindViewHolder(holderMain: RecyclerView.ViewHolder, position: Int) {
        when (holderMain.itemViewType) {
            1 -> {
                val holder = holderMain as ViewHolder
                holder.itemView.isClickable = false

                if (position == 0) {
                    holder.itemView.setOnClickListener {
                        MaterialAlertDialogBuilder(context, R.style.modeAlert)
                            .setTitle(context.getString(R.string.timer))
                            .setMessage(context.getString(R.string.setTimer) + SecondActivity.name)
                            .setPositiveButton(
                                context.getString(R.string.cont)
                            ) { _, _ ->
                                val intent = Intent(ACTION_SET_TIMER)
                                intent.putExtra(
                                    EXTRA_LENGTH,
                                    infoList[position].value.split(" ")[0].toInt() * 60
                                )
                                intent.putExtra(EXTRA_SKIP_UI, false)
                                intent.putExtra(EXTRA_MESSAGE, SecondActivity.name)
                                context.startActivity(intent)
                            }
                            .setNegativeButton(context.getString(R.string.cancel), null)
                            .show()
                    }
                }

                if (infoList[position].value == "0") holder.itemView.visibility = GONE
                holder.name.text = infoList[position].name
                holder.value.text = infoList[position].value
                holder.image.setImageResource(infoList[position].image)
                if (infoList[position].name == context.getString(R.string.source) && infoList[position].value != "Авторский") {
                    holder.itemView.setOnClickListener {
                        MaterialAlertDialogBuilder(context, R.style.modeAlert)
                            .setTitle(context.getString(R.string.redirect))
                            .setMessage(context.getString(R.string.redirectMessage))
                            .setPositiveButton(
                                context.getString(R.string.cont)
                            ) { _, _ ->
                                val url = "http://" + infoList[position].value
                                val browserIntent = Intent(Intent.ACTION_VIEW, Uri.parse(url))
                                context.startActivity(browserIntent)
                            }
                            .setNegativeButton(context.getString(R.string.cancel), null)
                            .show()
                    }
                } else if (infoList[position].value == "Авторский") {
                    holder.image.setImageResource(R.drawable.ic_round_edit_24)
                }
            }
            2 -> {
                val holder = holderMain as ViewHolder2

                holder.itemView.isClickable = false

                if (position == 0) {
                    holder.itemView.setOnClickListener {
                        MaterialAlertDialogBuilder(context, R.style.modeAlert)
                            .setTitle(context.getString(R.string.timer))
                            .setMessage(context.getString(R.string.setTimer) + SecondActivity.name)
                            .setPositiveButton(
                                context.getString(R.string.cont)
                            ) { _, _ ->
                                val intent = Intent(ACTION_SET_TIMER)
                                intent.putExtra(
                                    EXTRA_LENGTH,
                                    infoList[position].value.split(" ")[0].toInt() * 60
                                )
                                intent.putExtra(EXTRA_SKIP_UI, false)
                                intent.putExtra(EXTRA_MESSAGE, SecondActivity.name)
                                context.startActivity(intent)
                            }
                            .setNegativeButton(context.getString(R.string.cancel), null)
                            .show()
                    }
                }

                holder.name.text = infoList[position].name
                holder.value.text = infoList[position].value
                holder.image.setImageResource(infoList[position].image)
                if (infoList[position].name == context.getString(R.string.source) && infoList[position].value != "Авторский") {
                    holder.itemView.setOnClickListener {
                        MaterialAlertDialogBuilder(context, R.style.modeAlert)
                            .setTitle(context.getString(R.string.redirect))
                            .setMessage(context.getString(R.string.redirectMessage))
                            .setPositiveButton(
                                context.getString(R.string.cont)
                            ) { _, _ ->
                                val url = "http://" + infoList[position].value
                                val browserIntent = Intent(Intent.ACTION_VIEW, Uri.parse(url))
                                context.startActivity(browserIntent)
                            }
                            .setNegativeButton(context.getString(R.string.cancel), null)
                            .show()
                    }
                } else if (infoList[position].value == "Авторский") {
                    holder.image.setImageResource(R.drawable.ic_round_edit_24)
                }
                if (position == 7) {
                    val cursor: Cursor = mDb.rawQuery(
                        "SELECT * FROM recipes WHERE id = ?",
                        listOf(infoList[position].value).toTypedArray()
                    )
                    cursor.moveToFirst()
                    val starred = cursor.getInt(7) == 1
                    val banned = cursor.getInt(14) == 1
                    if (starred && banned) {
                        holder.banLayout.visibility = VISIBLE
                        holder.starLayout.visibility = VISIBLE
                        holder.itemView.visibility = VISIBLE
                        holder.value.text = context.getString(R.string.inStarred)
                        holder.value2.text = context.getString(R.string.inBanned)
                    } else if (starred && !banned) {
                        holder.value.text = context.getString(R.string.inStarred)
                        holder.banLayout.visibility = GONE
                        holder.starLayout.visibility = VISIBLE
                        holder.itemView.visibility = VISIBLE
                    } else if (!starred && banned) {
                        holder.starLayout.visibility = GONE
                        holder.banLayout.visibility = VISIBLE
                        holder.itemView.visibility = VISIBLE
                        holder.value2.text = context.getString(R.string.inBanned)
                    } else if (!starred && !banned) holder.itemView.visibility = GONE
                    cursor.close()
                }
            }
        }
    }

    override fun getItemCount(): Int {
        return infoList.size
    }

    inner class ViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        val name: TextView = view.findViewById(R.id.name)
        val value: TextView = view.findViewById(R.id.value)
        val image: ImageView = view.findViewById(R.id.image)
    }

    inner class ViewHolder2(view: View) : RecyclerView.ViewHolder(view) {
        val name: TextView = view.findViewById(R.id.name)
        val value: TextView = view.findViewById(R.id.value)
        val image: ImageView = view.findViewById(R.id.star)
        val value2: TextView = view.findViewById(R.id.value2)
        val starLayout: LinearLayout = view.findViewById(R.id.star_layout)
        val banLayout: LinearLayout = view.findViewById(R.id.ban_layout)
    }

    override fun onViewRecycled(holder: RecyclerView.ViewHolder) {
        super.onViewRecycled(holder)
        holder.itemView.setOnClickListener {}
    }

}